notebooks/ch12/LR_Case_Study.md

```diff
@@ -0,0 +1,792 @@
+      <th></th>
+      <th>Bias</th>
+      <th>Urban population (percentage)</th>
+      <th>Late births (reciprocal * 100)</th>
+      <th>Wine consumption per capita</th>
+      <th>Liquor consumption per capita</th>
+      <th>Cirrhosis death rate</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>0</th>
+      <td>1</td>
+      <td>44.0</td>
+      <td>33.2</td>
+      <td>5</td>
+      <td>30</td>
+      <td>41.2</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>1</td>
+      <td>43.0</td>
+      <td>33.8</td>
+      <td>4</td>
+      <td>41</td>
+      <td>31.7</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>1</td>
+      <td>48.0</td>
+      <td>40.6</td>
+      <td>3</td>
+      <td>38</td>
+      <td>39.4</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>1</td>
+      <td>52.0</td>
+      <td>39.2</td>
+      <td>7</td>
+      <td>48</td>
+      <td>57.5</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>1</td>
+      <td>71.0</td>
+      <td>45.5</td>
+      <td>11</td>
+      <td>53</td>
+      <td>74.8</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+Although we cannot see any null values from the first 5 rows of the dataset, there may be some present in our dataset that we will need to deal with. We can write code to loop through every column of the dataset and tell us if there are null values present using the pandas `isnull` function.
+
+
+```python
+for cname in drinking_data.columns.values:
+    column = drinking_data[cname] # Get the column corresponding to that column name
+    null_mask = column.isnull() # Get a mask that is True if there is a null value present, False otherwise
+    nulls_series = column[null_mask] # Get only values where the null value was detected
+    if len(nulls_series) > 0: # If a null value was detected
+        print(nulls_series)
+```
+
+    35   NaN
+    Name: Urban population (percentage), dtype: float64
+    
+
+It appears that there is a null value present in row 35 of our dataset in the `Urban population (percentage)` column. Let's examine that row more closely:
+
+
+```python
+drinking_data.iloc[35]
+```
+
+
+
+
+    Bias                                1.0
+    Urban population (percentage)       NaN
+    Late births (reciprocal * 100)     53.0
+    Wine consumption per capita        28.0
+    Liquor consumption per capita     149.0
+    Cirrhosis death rate              122.5
+    Name: 35, dtype: float64
+
+
+
+With null values, we have two options: either drop the row with the null value or try to impute a value in the place of the null. Since we only have one null value in our dataset and it may not be a good idea to try to guess a value because it may negatively influence our linear regression model, we can just drop the row.
+
+## Data Cleaning
+
+Let's rename the column names to shorter names. It will make our code look cleaner and easier to read. We can make a new list of column names and assign `drinking_data.columns` to this list.
```

Comment:
Good practice, but I'm not sure if the focus of this is the end-to-end case study or the linear regression

Reply:
It's more a part of the case study, working with data in general